### ISSUE TITLE
User can't auth with postgresql stored creds

### ISSUE THREAD

**Issue by LevBravE** [erlio/vernemq#519]
### Environment

- VerneMQ Version: 1.2.0
- OS: Ubuntu 16.04.2 LTS
- Postgresql: 9.6.2 

### Expected behavior

To authenticate an mqtt user with creds stored in postgresql.

### Actual behaviour

- Create table

```sql
CREATE EXTENSION pgcrypto;
CREATE TABLE vmq_auth_acl
 (
   mountpoint character varying(10) NOT NULL,
   client_id character varying(128) NOT NULL,
   username character varying(128) NOT NULL,
   password character varying(128),
   publish_acl json,
   subscribe_acl json,
   CONSTRAINT vmq_auth_acl_primary_key PRIMARY KEY (mountpoint, client_id, username)
 );
```
- Adding user creds to postgresql

```sql
WITH x AS (
    SELECT
        ''::text AS mountpoint,
           'test-station-FU01'::text AS client_id,
           'test-station-FU01'::text AS username,
           '123'::text AS password,
           gen_salt('bf', 10)::text AS salt,
           '[{"pattern": "test"}]'::json AS publish_acl,
           '[{"pattern": "test"}]'::json AS subscribe_acl
    )
INSERT INTO vmq_auth_acl (mountpoint, client_id, username, password, publish_acl, subscribe_acl)
    SELECT
        x.mountpoint,
        x.client_id,
        x.username,
        crypt(x.password, x.salt),
        publish_acl,
        subscribe_acl
    FROM x;
```

Result:
```
> SELECT * FROM vmq_auth_acl;

mountpoint |     client_id     |     username      |                           password                           |      publish_acl      |     subscribe_acl     
------------+-------------------+-------------------+--------------------------------------------------------------+-----------------------+-----------------------
            | test-station-FU01 | test-station-FU01 | $2a$10$/86Fw5NJ6rlOCadkoLyZXOaEmhyzK5iFlrWf1TNqcKCFqhup/lt3C | [{"pattern": "test"}] | [{"pattern": "test"}]

```

- Connecting using python script
```python
import paho.mqtt.client as mqtt

# The callback for when the client receives a CONNACK response from the server.
def on_connect(client, userdata, flags, rc):
    print("Connected with result code "+str(rc))

client = mqtt.Client(client_id="test_station_FU01")
client.username_pw_set("test_station_FU01", "321")
client.on_connect = on_connect

client.connect("<broker mqtt>", 1883)
client.loop_forever()
```

Result:
```
Connected with result code 5
```
- Verne mq has the following messages in logs

`console.log`
```
[debug] <0.221.0>@plumtree_broadcast:schedule_lazy_tick:682 0ms mailbox traversal, schedule next lazy broadcast in 10000ms, the min interval is 10000ms
[warning] <0.467.0>@vmq_mqtt_fsm:check_user:544 can't authenticate client {[],<<"test_station_FU01">>} due to error
[debug] <0.467.0>@vmq_ranch:teardown:123 session normally stopped
```

`error.log`
```
none
```

Could you please give me a hint what's wrong here?

Thanks!

**Comment by ioolkos** [comment]
Thanks @LevBravE for reporting.
I guess this should work...
I can think of 2 possible problems:
- check that the plugin(s) are correctly enabled. (you can send that part of your vernemq.conf, if you like)
- a problem with the way you generate the password (pgcrypto) maybe?

Anyone with more ideas?

**Comment by LevBravE** [comment]
Hi @ioolkos 
- Check vernemq.conf
```
plugins.vmq_diversity = on

plugins.vmq_passwd = off
plugins.vmq_acl = off

vmq_diversity.auth_postgres.enabled = on
...
```

- Generate the password (pgcrypto) - I checked it as follows
I copied the generated password in PostgreSQL to Redis. Plug in redis plugin and everything worked

**Comment by ioolkos** [comment]
Ok, thanks, I see that you follow the docs closely.

@dergraf have we changed any vmq_diversity config stuff that would make the docs obsolete?

**Comment by dergraf** [comment]
Is your password cprrect? 321 vs. 123?

**Comment by LevBravE** [comment]
`$2a$10$/86Fw5NJ6rlOCadkoLyZXOaEmhyzK5iFlrWf1TNqcKCFqhup/lt3C = 321`

**Comment by dergraf** [comment]
Is your client id correct? unserscore vs. dashes... 
Sorry for asking those questions, but given your example code and SQL output, it has to complain as the password is `123` and the clientid & username are using dashes in the database but underscores in your client code.

**Comment by LevBravE** [comment]
Exactly how I did not notice it ....

Thank you!)

**Comment by Safari94** [comment]
Hi,

I am facing the same issue :( 
**### I am using:**
->Vernemq clone from repo
->Ubuntu 20.04 LTS
->PostgreSQL 12.2 (Ubuntu 12.2-4) 

**### Logs from console.log**


vermemq.conf

![image](https://user-images.githubusercontent.com/11853732/83412671-70f9b100-a412-11ea-87ca-fa50ec9400bc.png)

![image](https://user-images.githubusercontent.com/11853732/83412719-88d13500-a412-11ea-8cb2-b2793e51e3fd.png)

![image](https://user-images.githubusercontent.com/11853732/83412823-b3bb8900-a412-11ea-997c-d8368e42f4c6.png)

![image](https://user-images.githubusercontent.com/11853732/83412845-bd44f100-a412-11ea-94df-84bc08be4ad1.png)

I believe connection with DB is set by console.log
2020-06-01 15:19:19.239 [info] <0.5949.0> Administrative stop
2020-06-01 15:19:30.936 [info] <0.204.0>@plumtree_peer_service_manager:maybe_load_state_from_disk:131 read state from file {[{[{actor,<<81,104,232,90,238,242,17,14>
2020-06-01 15:19:30.937 [info] <0.204.0>@plumtree_peer_service_manager:write_state_to_disk:100 writing state {[{[{actor,<<81,104,232,90,238,242,17,148,245,198,181,>
2020-06-01 15:19:31.138 [info] <0.209.0>@plumtree_metadata_leveldb_instance:init_state:322 Datadir ./data/meta/meta/0 options for LevelDB: [{open,[{block_cache_thr>
2020-06-01 15:19:31.335 [info] <0.213.0>@plumtree_metadata_leveldb_instance:init_state:322 Datadir ./data/meta/meta/1 options for LevelDB: [{open,[{block_cache_thr>
2020-06-01 15:19:31.433 [info] <0.214.0>@plumtree_metadata_leveldb_instance:init_state:322 Datadir ./data/meta/meta/2 options for LevelDB: [{open,[{block_cache_thr>
2020-06-01 15:19:31.534 [info] <0.215.0>@plumtree_metadata_leveldb_instance:init_state:322 Datadir ./data/meta/meta/3 options for LevelDB: [{open,[{block_cache_thr>
2020-06-01 15:19:31.542 [info] <0.216.0>@plumtree_metadata_leveldb_instance:init_state:322 Datadir ./data/meta/meta/4 options for LevelDB: [{open,[{block_cache_thr>
2020-06-01 15:19:31.554 [info] <0.217.0>@plumtree_metadata_leveldb_instance:init_state:322 Datadir ./data/meta/meta/5 options for LevelDB: [{open,[{block_cache_thr>
2020-06-01 15:19:31.655 [info] <0.218.0>@plumtree_metadata_leveldb_instance:init_state:322 Datadir ./data/meta/meta/6 options for LevelDB: [{open,[{block_cache_thr>
2020-06-01 15:19:31.736 [info] <0.219.0>@plumtree_metadata_leveldb_instance:init_state:322 Datadir ./data/meta/meta/7 options for LevelDB: [{open,[{block_cache_thr>
2020-06-01 15:19:31.750 [info] <0.220.0>@plumtree_metadata_leveldb_instance:init_state:322 Datadir ./data/meta/meta/8 options for LevelDB: [{open,[{block_cache_thr>
2020-06-01 15:19:31.855 [info] <0.221.0>@plumtree_metadata_leveldb_instance:init_state:322 Datadir ./data/meta/meta/9 options for LevelDB: [{open,[{block_cache_thr>
2020-06-01 15:19:31.936 [info] <0.222.0>@plumtree_metadata_leveldb_instance:init_state:322 Datadir ./data/meta/meta/10 options for LevelDB: [{open,[{block_cache_th>
2020-06-01 15:19:32.036 [info] <0.223.0>@plumtree_metadata_leveldb_instance:init_state:322 Datadir ./data/meta/meta/11 options for LevelDB: [{open,[{block_cache_th>
2020-06-01 15:19:32.603 [info] <0.198.0>@vmq_metadata:start:27 Try to start vmq_plumtree: ok
2020-06-01 15:19:33.042 [info] <0.198.0>@vmq_message_store:start:27 Try to start vmq_generic_msg_store: ok
2020-06-01 15:19:33.229 [info] <0.325.0>@vmq_reg_trie:handle_info:204 loaded 0 subscriptions into vmq_reg_trie
2020-06-01 15:19:33.243 [info] <0.225.0>@vmq_cluster:init:141 cluster event handler 'vmq_cluster' registered
2020-06-01 15:19:34.747 [error] <0.375.0>@vmq_mqtt_fsm:check_user:563 can't authenticate client {[],<<"test-station-FU011591008497768">>} from 79.168.247.15:55495 >
2020-06-01 15:19:35.701 [info] <0.424.0>@vmq_diversity_app:start:71 enable auth script for postgres "./share/lua/auth/postgres.lua"

What am I doing wrong?

**Comment by ioolkos** [comment]
@Safari94 I don't know what's wrong here, the plugin itself seems to be working. It tells you it can't authenticate the client. Double-check everything. You seem to partially copy/paste from the previous user (issue opener), so double check that too.

**Comment by Safari94** [comment]
@ioolkos I already did that. In order to create the DB I used the documentation provided on VerneMQ website.
![image](https://user-images.githubusercontent.com/11853732/83424130-e8840c00-a423-11ea-8810-88d8178f13e9.png)
Already remove the pgcrypto extension and creat it again.
I am testing using MQTTbox and the script form the issue opener.

![image](https://user-images.githubusercontent.com/11853732/83424286-1ec18b80-a424-11ea-87cb-d9240b3c521e.png)

![image](https://user-images.githubusercontent.com/11853732/83424370-37ca3c80-a424-11ea-949c-65d8a6fedab6.png)

No luck :(
